packages/client/hmi-client/src/services/workflow.ts

```diff
@@ -473,8 +473,29 @@ export class WorkflowWrapper {
 				const targetPort = targetNode.inputs.find((port) => port.id === edge.targetPortId);
 				if (!targetPort) return;
 				edge.sourcePortId = selected.id; // Sync edge source port to selected output
+
+				const selectedTypes = selected.type.split('|').map((d) => d.trim());
+				const allowedInputTypes = targetPort.type.split('|').map((d) => d.trim());
+				const intersectionTypes = _.intersection(selectedTypes, allowedInputTypes);
+
+				// Not supported if there are more than one match
+				if (intersectionTypes.length > 1) {
+					console.error(`Ambiguous matching types [${selectedTypes}] to [${allowedInputTypes}]`);
+					return;
+				}
+				if (intersectionTypes.length === 0) {
+					return;
+				}
+				if (selectedTypes.length > 1) {
```

Comment:
nitpick: may want to add comment with #intersectionTypes is implicitly length 1

Reply:
added a bit more comments

Reply:
Thanks!